### PR TITLE
geo/wkt: add support for parsing multilinestrings with Z and M dims

### PR DESCRIPTION
**Pull request by andyyang890**
This patch extends the capabilities of the WKT parser to include
parsing of multilinestrings with Z and M dimensions.

Refs: #53091

Release note: None

### CODE REVIEW

File: pkg/geo/wkt/wkt.y [hunk 1]
```diff
@@ -41,14 +41,18 @@ type geomPair struct {
 	ends       []int
 }
 
+func makeGeomPair(flatCoords []float64) geomPair {
```
Comment (otan) [inline pkg/geo/wkt/wkt.y:44]: let's call `geomPair` something else. maybe `geomFlatCoordsRepr` or something. the pair terminology is getting to me.

**Review by otan — APPROVED** [review]

**Comment by andyyang890** [comment]
bors r=otan

**Comment by andyyang890** [comment]
bors r=otan

**Comment by andyyang890** [comment]
bors r=otan

**Pull request merged**